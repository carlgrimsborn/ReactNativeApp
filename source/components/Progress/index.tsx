import * as React from 'react';
import { View, Text, ActivityIndicator } from 'react-native';

const Progress = ({amount}: IProps) => {
  return (
    <View>
      <ActivityIndicator size='small' color='#0000ff' />
      <Text>Progress: {amount}</Text>
    </View>
  );
};

interface IProps {
  amount: number;
}

export default Progress;
