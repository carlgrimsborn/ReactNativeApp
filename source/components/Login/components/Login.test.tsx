
/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';
import Progress from '../../Progress';
import { TextInput } from 'react-native';

import { Login } from './index';

jest.unmock('react-native');

function setup() {
  const props = {
// Här kan ni mocka komponentens props om sådana finns
// typ userName: 'test'
  };

  const enzymeWrapper = shallow(<Login {...props} /> );

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *   import { Text } from 'react-native';
   *    const busyProps = enzymeWrapper.find(Text).props();
   *    expect(busyProps.text).toBe('test);
   */
  describe('Login', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
      const textOutput1 = enzymeWrapper.find(TextInput).at(0).props();
      expect(textOutput1.placeholder).toBe('Username');

      const textOutput2 = enzymeWrapper.find(TextInput).at(1).props();
      expect(textOutput2.placeholder).toBe('Password');

    });
  });
});