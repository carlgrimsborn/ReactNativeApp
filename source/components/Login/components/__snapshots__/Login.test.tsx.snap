// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Login
        </Text>,
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          keyboardType="email-address"
          placeholder="Username"
          style={
            Object {
              "width": 200,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "width": 200,
            }
          }
        />,
      ],
      "style": Object {
        "flex": 1,
        "padding": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Login",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoFocus": true,
          "keyboardType": "email-address",
          "placeholder": "Username",
          "style": Object {
            "width": 200,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "onChangeText": [Function],
          "placeholder": "Password",
          "secureTextEntry": true,
          "style": Object {
            "width": 200,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Login
          </Text>,
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            keyboardType="email-address"
            placeholder="Username"
            style={
              Object {
                "width": 200,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              Object {
                "width": 200,
              }
            }
          />,
        ],
        "style": Object {
          "flex": 1,
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Login",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoFocus": true,
            "keyboardType": "email-address",
            "placeholder": "Username",
            "style": Object {
              "width": 200,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "onChangeText": [Function],
            "placeholder": "Password",
            "secureTextEntry": true,
            "style": Object {
              "width": 200,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
